app/src/main/java/com/connorlay/knitgrid/activities/PatternDetailActivity.java

```diff
@@ -3,29 +3,55 @@
 import android.os.Bundle;
 import android.support.v4.app.FragmentTransaction;
 import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.widget.Button;
 
 import com.connorlay.knitgrid.R;
+import com.connorlay.knitgrid.fragments.PatternGridLegendFragment;
 import com.connorlay.knitgrid.fragments.PatternHighlightFragment;
 import com.connorlay.knitgrid.models.Pattern;
 import com.connorlay.knitgrid.presenters.PatternPresenter;
 
 import butterknife.ButterKnife;
 
-public class PatternDetailActivity extends AppCompatActivity {
+public class PatternDetailActivity extends AppCompatActivity implements View.OnClickListener {
     public static final String STITCH_EXTRA = "STITCH_EXTRA";
     public static final String ARG_PATTERN = "PatternDetailActivity.ArgPattern";
+    private PatternPresenter patternPresenter;
+    private Pattern pattern;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_pattern_detail_view);
         ButterKnife.bind(this);
 
-        Pattern pattern = getIntent().getParcelableExtra(ARG_PATTERN);
-        PatternPresenter patternPresenter = new PatternPresenter(pattern);
-        FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
+        pattern = getIntent().getParcelableExtra(ARG_PATTERN);
```

Comment:
Oh ok I see. In that case I would leave `pattern` as a field and remove `patternPresenter` as a field and just initialize it in `onClick`.